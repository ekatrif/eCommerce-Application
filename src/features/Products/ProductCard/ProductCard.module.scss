.item {
  width: calc((100% - 4em) / 3);
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: white;
  padding: 1.5em;
  height: 43em;
  overflow: hidden;
  transition: all 0.5s ease-in;

  &:hover {
    cursor: pointer;
    box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.4), -2px -2px 4px rgba(0, 0, 0, 0.4);
    transition: all 0.5s ease-out;
  }

  & > * + * {
    margin-top: 1em;
  }

  @media (max-width: 1220px) {
    & {
      max-width: 45%;
    }
  }

  @media (max-width: 768px) {
    & {
      max-width: 100%;
    }
  }
}

.title {
  font-weight: 700;
  font-size: 1.4rem;
}

.prices {
  display: flex;
  gap: 1em;
}

.price {
  font-size: 1.5rem;
  font-weight: 500;
  color: inherit;
}

.discount {
  font-size: 1.5rem;
  color: #e5e5e5;
  text-decoration: line-through;
}

.image {
  width: 100%;
  object-fit: cover;
}

.description {
  margin-top: 2em;
}