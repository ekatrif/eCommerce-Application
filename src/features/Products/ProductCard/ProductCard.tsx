import React from 'react';
import { Languages } from '../../../api/types';
import formatPrice from '../../../utils/catalog/format-price';
import truncateString from '../../../utils/catalog/truncate-string';
import classes from './ProductCard.module.scss';
import Button from '../../../components/UI/button/Button';
import { IResult } from '../types';

interface IProductProps {
  product: IResult;
  key?: string;
}

const ProductCard: React.FC<IProductProps> = ({ product }) => {
  return (
    <li className={classes.item}>
      <h3>
        <a
          href={`catalog/${product.id}`}
          title={product.masterData.staged.name[Languages.English]}
          className={classes.title}
        >
          {product.masterData.staged.name[Languages.English]}
        </a>
      </h3>
      {product.masterData.staged.masterVariant.prices[0].discounted ? (
        <div className={classes.prices}>
          <span className={classes.discount}>
            {formatPrice(product.masterData.staged.masterVariant.prices[0].value.centAmount)}{' '}
            {product.masterData.staged.masterVariant.prices[0].value.currencyCode}
          </span>
          <span className={classes.price}>
            {formatPrice(product.masterData.staged.masterVariant.prices[0].discounted.value.centAmount)}{' '}
            {product.masterData.staged.masterVariant.prices[0].discounted.value.currencyCode}
          </span>
        </div>
      ) : (
        <span className={classes.price}>
          {formatPrice(product.masterData.staged.masterVariant.prices[0].value.centAmount)}{' '}
          {product.masterData.staged.masterVariant.prices[0].value.currencyCode}
        </span>
      )}
      <a href="#" title={product.masterData.staged.name[Languages.English]} className={classes.link}>
        <img
          src={product.masterData.staged.masterVariant.images[0].url}
          alt={product.masterData.staged.name[Languages.English]}
          className={classes.image}
        />
      </a>
      <Button type="button" text="Add to cart" />
      <p>
        <a href="#" title={product.masterData.staged.name[Languages.English]} className={classes.link}>
          {truncateString(product.masterData.staged.description[Languages.English], 200)}
        </a>
      </p>
    </li>
  );
};

export default ProductCard;
