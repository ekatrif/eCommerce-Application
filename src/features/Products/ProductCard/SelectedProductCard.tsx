import React from 'react';
import { Languages } from '../../../api/types';
import formatPrice from '../../../utils/catalog/format-price';
import truncateString from '../../../utils/catalog/truncate-string';
import classes from './ProductCard.module.scss';
import Button from '../../../components/UI/button/Button';
import { ISelectedProduct } from '../types';

interface IProductProps {
  product: ISelectedProduct;
  key?: string;
}

const SelectedProductCard: React.FC<IProductProps> = ({ product }) => {
  if (product) {
    const productName = product.name[Languages.English];
    const productDescription = product.description[Languages.English];
    const productRegularPrice = product.masterVariant.prices[0].value.centAmount;
    const productDiscountedPrice = product.masterVariant.prices[0].discounted;
    const productImage = product.masterVariant.images[0].url;

    return (
      <li className={classes.item}>
        <h3>
          <a href={`catalog/${product.id}`} title={productName} className={classes.title}>
            {productName}
          </a>
        </h3>
        {productDiscountedPrice ? (
          <div className={classes.prices}>
            <span className={classes.discount}>
              {formatPrice(productRegularPrice)}
              {'  $'}
            </span>
            <span className={classes.price}>
              {formatPrice(productDiscountedPrice.value.centAmount)} {productDiscountedPrice.value.currencyCode}
            </span>
          </div>
        ) : (
          <span className={classes.price}>
            {formatPrice(productRegularPrice)}
            {' $'}
          </span>
        )}
        <a href="#" title={productName} className={classes.link}>
          <img src={productImage} alt={productName} className={classes.image} />
        </a>
        <Button type="button" text="Add to cart" />
        <p>
          <a href="#" title={productName} className={classes.link}>
            {truncateString(productDescription, 200)}
          </a>
        </p>
      </li>
    );
  }

  return null;
};

export default SelectedProductCard;
