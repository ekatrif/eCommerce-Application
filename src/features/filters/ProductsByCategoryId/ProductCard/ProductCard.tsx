import React from 'react';
import { Languages } from '../../../../api/types';
import formatPrice from '../../../../utils/catalog/format-price';
import truncateString from '../../../../utils/catalog/truncate-string';
import classes from '../../../Products/ProductCard/ProductCard.module.scss';
import Button from '../../../../components/UI/button/Button';
import { IResult } from '../types';

interface IProductProps {
  product: IResult;
  key?: string;
}

const ProductCard: React.FC<IProductProps> = ({ product }) => {
  return (
    <li className={classes.item}>
      <h3>
        <a href="#" title={product.name[Languages.English]} className={classes.title}>
          {product.name[Languages.English]}
        </a>
      </h3>
      {product.masterVariant.prices[0].discounted ? (
        <div className={classes.prices}>
          <span className={classes.discount}>
            {formatPrice(product.masterVariant.prices[0].value.centAmount)}{' '}
            {product.masterVariant.prices[0].value.currencyCode}
          </span>
          <span className={classes.price}>
            {formatPrice(product.masterVariant.prices[0].discounted.value.centAmount)}{' '}
            {product.masterVariant.prices[0].discounted.value.currencyCode}
          </span>
        </div>
      ) : (
        <span className={classes.price}>
          {formatPrice(product.masterVariant.prices[0].value.centAmount)}{' '}
          {product.masterVariant.prices[0].value.currencyCode}
        </span>
      )}
      <a href="#" title={product.name[Languages.English]} className={classes.link}>
        <img
          src={product.masterVariant.images[0].url}
          alt={product.name[Languages.English]}
          className={classes.image}
        />
      </a>
      <Button type="button" text="Add to cart" />
      <p>
        <a href="#" title={product.name[Languages.English]} className={classes.link}>
          {truncateString(product.description[Languages.English], 200)}
        </a>
      </p>
    </li>
  );
};

export default ProductCard;
