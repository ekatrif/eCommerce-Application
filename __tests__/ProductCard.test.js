import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../src/features/Products/ProductCard/ProductCard';

test('ProductCard renders correctly', () => {
  const tree = renderer
    .create(<ProductCard />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
