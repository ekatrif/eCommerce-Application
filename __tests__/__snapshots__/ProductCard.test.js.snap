// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `null`;
